xapian-core/cluster/cluster.cc

```diff
@@ -0,0 +1,179 @@
+{
+    LOGCALL_CTOR(API, "MSetDocumentSource", mset | maxitems);
+    if (maxitems > mset.size()) {
+		maxitems = mset.size();
+    }
+
+    if (maxitems > 0) {
+		mset.fetch(mset.begin(), mset[maxitems - 1]);
+    }
+}
+
+Document
+MSetDocumentSource::next_document() {
+	LOGCALL(API, Document, "MSetDocumentSource::next_document()", "");
+	return (mset[index++].get_document());
+}
+
+bool
+MSetDocumentSource::at_end() const {
+	LOGCALL(API, int, "MSetDocumentSource::at_end()", "");
+	return (index >= maxitems);
+}
+
+doccount
+MSetDocumentSource::size() {
+	return mset.size();
+}
+
+Similarity::~Similarity()
+{
+	LOGCALL_DTOR(API, "Similarity");
+}
+
+FreqSource::~FreqSource()
+{
+    LOGCALL_DTOR(API,"FreqSource");
+}
+
+doccount
+DummyFreqSource::get_termfreq(const string &tname) {
+	LOGCALL(API, doccount, "DummyFreqSource::get_termfreq()", "");
+	return 1;
+}
+
+doccount
+DummyFreqSource::get_doccount() {
+	LOGCALL_VOID(API, doccount, "DummyFreqSource::get_doccount()");
+	return 1;
+}
+
+void
+TermListGroup::add_document(const Document &document) {
+    LOGCALL_VOID(API, "TermListGroup::add_document(Document &document)", document);
+    
+    TermIterator titer(document.termlist_begin());
+    TermIterator end(document.termlist_end());
+    
+    for(; titer != end; titer++)
+    {
+        map<string, doccount>::iterator i;
+        i = termfreq.find(*titer);
+        if(i == termfreq.end())
+        {
+            termfreq[*titer]=1;
+        }
+        else
+        {
+            i->second+=1;
+        }
+    }
+}
+
+void
+TermListGroup::add_documents(DocumentSource &docs) {
+	LOGCALL_VOID(API, "TermListGroup::add_documents(DocumentSource &docs)", docs);
+	while(!docs.at_end()) {
+		Document document = docs.next_document();
+		add_document(document);
+	}
+	docs_num = docs.size();
+}
+
+doccount
+TermListGroup::get_doccount() {
+	LOGCALL(API, doccount, "VectorTermIterator::get_doccount()","");
+	return docs_num;
+}
+    
+doccount
+TermListGroup::get_termfreq(const string &tname) {
+	LOGCALL(API, doccount, "VectorTermIterator::get_termfreq()","");
+	return termfreq[tname];
+}
+
+
+doccount
+ClusterSet::num_of_clusters() {
+	LOGCALL(API, doccount, "ClusterSet::num_of_clusters()","");
+	return clusters.size();
+}
+
+doccount
+ClusterSet::cluster_size(clusterid cid) {
+	LOGCALL(API, doccount, "ClusterSet::cluster_size(clusterid cid)", cid);
+	map<clusterid, vector<Document> >::iterator i;
+	i = clusters.find(cid);
+	//Add error handling here
+	if(i != clusters.end())
+		return ((i->second).size());	
+}
+
+ClusterSetIterator
+ClusterSet::begin() {
+	LOGCALL(API, ClusterSetIterator, "ClusterSet::begin()","");
+	return ClusterSetIterator(*this, 0);
+}
+
+ClusterSetIterator
+ClusterSet::end() {
+	LOGCALL(API, ClusterSetIterator, "ClusterSet::end()","");
+	return ClusterSetIterator(*this, num_of_clusters());
+}
+
+ClusterSetIterator
+ClusterSet::operator[](doccount i) {
+	return ClusterSetIterator(*this, i);
+}
+
+
+/*
```

Comment:
You shouldn't commit commented code, because it's not part of the "active" changes in this commit. If you use `git add -p` to selectively add changes in a file, it becomes easier to skip the things you don't want in a commit. (For completely new files, `git add -N` is "intent to add" the file, which means you can then do `git add -p`.)